alerts.js

```diff
@@ -0,0 +1,181 @@
+require('dotenv').config()
+const BN = require('bignumber.js')
+const Web3 = require('web3')
+const logger = require('./logger')('alerts')
+const eventsInfo = require('./utils/events')
+
+const { HOME_RPC_URL, FOREIGN_RPC_URL } = process.env
+
+const homeProvider = new Web3.providers.HttpProvider(HOME_RPC_URL)
+const web3Home = new Web3(homeProvider)
+
+const foreignProvider = new Web3.providers.HttpProvider(FOREIGN_RPC_URL)
+const web3Foreign = new Web3(foreignProvider)
+
+async function main() {
+  try {
+    const { foreignDeposits, homeDeposits, homeWithdrawals, foreignWithdrawals } = await eventsInfo()
+
+    const xSignatures = foreignDeposits.filter(findDifferences(homeDeposits))
+    const xAffirmations = homeWithdrawals.filter(findDifferences(foreignWithdrawals))
+
+    logger.debug("building misbehavior blocks")
+    const getBlockNumber = web3 => web3.eth.getBlockNumber()
+    const [ foreignBlockNumber, homeBlockNumber ] = (await Promise.all([web3Foreign, web3Home].map(getBlockNumber))).map(Web3.utils.toBN)
+
+    const xSignaturesMisbehavior = buildRangesObject(xSignatures.map(findMisbehaviorRange(foreignBlockNumber)).reduce(mergeRanges, []))
+    const xAffirmationsMisbehavior = buildRangesObject(xAffirmations.map(findMisbehaviorRange(homeBlockNumber)).reduce(mergeRanges, []))
+
+    logger.debug("extracting most recent transactionHash")
+    const { transactionHash: xSignaturesMostRecentTxHash = '' } = xSignatures.sort(sortEvents).reverse()[0] || {}
```

Comment:
We need to output content of `misbehavior` even if contains just falses:
```json
        "misbehavior": {
            "last60blocks": false,
            "last60to180blocks": false,
            "last180to720blocks": false,
            "last720to17280blocks": false,
            "last17280blocks": false
        },
``` 
instead of
```json
        "misbehavior": {},
```
This will be used by the alarm system to check if the corresponding value was changed from `false` to `true`.

Reply:
@akolotov, done